templates.go

```diff
@@ -0,0 +1,181 @@
+package postmark
+
+import (
+	"golang.org/x/net/context"
+	"net/url"
+	"strconv"
+)
+
+// Templates defines the functionality of the template resource
+type Templates interface {
+	// Get retrieves an individual template
+	// http://developer.postmarkapp.com/developer-api-templates.html#get-template
+	Get(ctx context.Context, id string) (*Template, error)
+
+	// Create creates a new template within Postmark
+	// http://developer.postmarkapp.com/developer-api-templates.html#create-template
+	Create(ctx context.Context, tmpl *Template) (*TemplateResp, error)
+
+	// Edit modifies an existing template
+	// http://developer.postmarkapp.com/developer-api-templates.html#edit-template
+	Edit(ctx context.Context, id string, tmpl *Template) (*TemplateResp, error)
+
+	// List returns a list of all existing templates
+	// http://developer.postmarkapp.com/developer-api-templates.html#template-list
+	List(ctx context.Context, count, offset int) (*TemplateList, error)
+
+	// Delete permanently deletes a template from Postmark
+	// http://developer.postmarkapp.com/developer-api-templates.html#delete-template
+	Delete(ctx context.Context, id string) (*TemplateResp, error)
+
+	// Validate allows a template's validity to be checked without sending the template
+	// http://developer.postmarkapp.com/developer-api-templates.html#validate-template
+	Validate(ctx context.Context, tmpl *TemplateValidation) (*TemplateValidationResp, error)
+}
+
+type templates struct {
+	pm *postmark
+}
+
+var _ Templates = (*templates)(nil)
+
+// Template defines the template entities within postmark
+type Template struct {
+	TemplateID         int64 `json:"TemplateId"`
+	Name               string
+	Subject            string
+	HTMLBody           string `json:"HtmlBody"`
+	TextBody           string
+	AssociatedServerID int64 `json:"AssociatedServerId"`
+	Active             bool
+}
+
+func (t *templates) Get(ctx context.Context, id string) (*Template, error) {
+	tmpl := new(Template)
+	_, err := t.pm.Exec(ctx, &Request{
+		Method: "GET",
+		Path:   "/templates/" + id,
+		Target: tmpl,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return tmpl, nil
+}
+
+// TemplateResp defines the response after template manipulation
+type TemplateResp struct {
```

Comment:
I think there are a few too many empty fields for it to make sense, primarily the html/text body fields. I think it's a bit confusing if those are always empty.